siriuspy/siriuspy/devices/scraper.py

```diff
@@ -0,0 +1,311 @@
+"""Storage Ring Scrapers."""
+
+from .device import Device as _Device
+
+
+class _ScraperDev(_Device):
+    """."""
+
+    class DEVICES:
+        """Devices names."""
+
+        H = 'SI-01SA:DI-ScrapH'
+        V = 'SI-01SA:DI-ScrapV'
+        ALL = (H, V, )
+
+    _properties = (
+        'Home-Cmd', 'ForceComplete-Mon', 'CoordConvErr-Mon',
+        'EnblBacklashComp-Sel', 'EnblBacklashComp-Sts'
+    )
+
+    def __init__(self, devname, properties=None):
+        """."""
+        # check if device exists
+        if devname not in _ScraperDev.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        if properties is None:
+            properties = _ScraperDev._properties
+
+        super().__init__(devname, properties)
+
+    @property
+    def is_force_cmd_complete(self):
+        """."""
+        return self['ForceComplete-Mon']
+
+    @property
+    def is_coordinate_convertion_ok(self):
+        """."""
+        return self['CoordConvErr-Mon']
+
+    @property
+    def is_backlash_compensation_enabled(self):
+        """."""
+        return self['EnblBacklashComp-Sts']
+
+    def cmd_enable_backlash_compensation(self):
+        """."""
+        self['EnblBacklashComp-Sel'] = 1
+
+    def cmd_disable_backlash_compensation(self):
+        """."""
+        self['EnblBacklashComp-Sel'] = 0
+
+    def cmd_go_max_aperture(self):
+        """Go to maximum slit aperture."""
+        self['Home-Cmd'] = 1
+
+
+class ScraperH(_ScraperDev):
+    """."""
+
+    _properties = (
+        'OuterSlitPos-SP', 'OuterSlitPos-RB',
+        'InnerSlitPos-SP', 'InnerSlitPos-RB',
+        'InnerDoneMov-Mon', 'OuterDoneMov-Mon',
+        'OuterSlitInnerLim-SP', 'OuterSlitInnerLim-RB',
+        'OuterSlitOuterLim-SP', 'OuterSlitOuterLim-RB',
+        'OuterSlitBacklashDist-SP', 'OuterSlitBacklashDist-RB',
+        'OuterMotionCtrl-Cte', 'ForceOuterSlitPos-Cmd',
+        'InnerSlitInnerLim-SP', 'InnerSlitInnerLim-RB',
+        'InnerSlitOuterLim-SP', 'InnerSlitOuterLim-RB',
+        'InnerSlitBacklashDist-SP', 'InnerSlitBacklashDist-RB',
+        'InnerMotionCtrl-Cte', 'ForceInnerSlitPos-Cmd',
+        )
+
+    def __init__(self):
+        """."""
+        # call base class constructor
+        _all_properties = _ScraperDev._properties + ScraperH._properties
+        super().__init__(
+            devname=_ScraperDev.DEVICES.H, properties=_all_properties)
+
+    @property
+    def left_slit(self):
```

Comment:
I think it would be more intuitive if these properties ended with a `_pos` to indicate that it returns the position of the slits.

Reply:
done!